modules/http4s/src/smithy4s/http4s/SmithyHttp4sRouter.scala

```diff
@@ -23,23 +23,38 @@ import cats.implicits._
 import org.http4s._
 import smithy4s.http4s.internals.SmithyHttp4sServerEndpoint
 import smithy4s.kinds._
+import org.typelevel.vault.Key
+import cats.Id
+import cats.Applicative
+import cats.effect.kernel.Unique
 
 // format: off
 class SmithyHttp4sRouter[Alg[_[_, _, _, _, _]], Op[_, _, _, _, _], F[_]](
     service: smithy4s.Service.Aux[Alg, Op],
     impl: FunctorInterpreter[Op, F],
     errorTransformation: PartialFunction[Throwable, F[Throwable]],
-    entityCompiler: EntityCompiler[F]
+    entityCompiler: EntityCompiler[F],
+    middleware: EndpointSpecificMiddleware[F]
 )(implicit effect: EffectCompat[F]) {
 
+  private val pathParamsKey = {
+    implicit val cheatUnique: Unique[Id] = new Unique[Id] {
+      def applicative: Applicative[Id] = Applicative[Id]
+
+      def unique: Id[Unique.Token] = new Unique.Token()
+
+    }
+    Key.newKey[Id, smithy4s.http.PathParams]
```

Comment:
http4s creates keys this way

```scala
val ConnectionInfo: Key[Connection] = Key.newKey[SyncIO, Connection].unsafeRunSync()
```

it's `SyncIO`, so it's fine to `.unsafeRunSync()` on any platform. I'd say let's do the same

Reply:
Meh, that's an abuse imho. My point of view is that http4s should not depend on cats-effect-core, but cats-effect-std (or even -kernel). I wish there was just an `unsafeCreate` on `Key`.

Reply:
yeah, it should not, but it does ;)

(I think it's because fs2 has an explicit SyncIO compiler)

> I wish there was just an `unsafeCreate` on Key.

you know what to do...

Reply:
> you know what to do... 

I do : tell @lewisjkl to ignore your PR comment and keep the current implementation that is totally fine  :trollface:

Reply:
Just know that the `new Unique` gives me the heebiejeebies :P

Reply:
Well, it may well be that we need to use `SyncIO` in CE2 :/